src/main/java/org/sonarsource/sonarlint/ls/backend/BackendService.java

```diff
@@ -36,32 +38,54 @@
 import org.sonarsource.sonarlint.core.clientapi.backend.connection.config.SonarCloudConnectionConfigurationDto;
 import org.sonarsource.sonarlint.core.clientapi.backend.connection.config.SonarQubeConnectionConfigurationDto;
 import org.sonarsource.sonarlint.core.clientapi.backend.hotspot.OpenHotspotInBrowserParams;
-import org.sonarsource.sonarlint.core.clientapi.backend.rules.GetActiveRuleDetailsParams;
-import org.sonarsource.sonarlint.core.clientapi.backend.rules.GetActiveRuleDetailsResponse;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.GetEffectiveRuleDetailsParams;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.GetEffectiveRuleDetailsResponse;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.GetStandaloneRuleDescriptionParams;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.GetStandaloneRuleDescriptionResponse;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.ListAllStandaloneRulesDefinitionsResponse;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.UpdateStandaloneRulesConfigurationParams;
 import org.sonarsource.sonarlint.ls.connected.ProjectBindingWrapper;
 import org.sonarsource.sonarlint.ls.settings.ServerConnectionSettings;
 
 public class BackendService {
 
   private final SonarLintBackend backend;
+  private final CountDownLatch initializeLatch = new CountDownLatch(1);
 
   public BackendService(SonarLintBackend backend) {
     this.backend = backend;
   }
 
   public void initialize(InitializeParams backendInitParams) {
-    backend.initialize(backendInitParams);
+    try {
+      backend.initialize(backendInitParams).thenRun(initializeLatch::countDown).get();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException("Interrupted", e);
+    } catch (ExecutionException e) {
+      throw new IllegalStateException("Could not initialize SonarLint Backend", e);
+    }
+  }
+
+  private SonarLintBackend initializedBackend() {
+    try {
+      initializeLatch.await();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException("SonarLint backend initialization interrupted", e);
+    }
+    return backend;
   }
 
   public void openHotspotInBrowser(OpenHotspotInBrowserParams params) {
-    backend.getHotspotService().openHotspotInBrowser(params);
+    initializedBackend().getHotspotService().openHotspotInBrowser(params);
```

Comment:
we need to make sure that the backend is properly initialized before making any further requests